packages/mosaic/sql/src/visit/codegen/duckdb.ts

```diff
@@ -0,0 +1,439 @@
+      + ' END';
+  }
+
+  visitCast(node: CastNode): string {
+    const { expr, cast } = node;
+    return `(${this.toString(expr)})::${cast}`;
+  }
+
+  visitCollate(node: CollateNode): string {
+    const { expr, collation } = node;
+    return `${this.toString(expr)} COLLATE ${collation}`;
+  }
+
+  visitColumnParam(node: ColumnParamNode): string {
+    return this.visitColumnRef(node);
+  }
+
+  visitColumnRef(node: ColumnRefNode): string {
+    const { column, table } = node;
+    const tref = table ? `${this.toString(table)}.` : '';
+    const id = column === '*' ? '*' : quoteIdentifier(column);
+    return `${tref}${id}`;
+  }
+
+  visitDescribeQuery(node: DescribeQuery): string {
+    const { query } = node;
+    return `DESCRIBE ${this.toString(query)}`;
+  }
+
+  visitExpression(node: ExprNode): string {
+    // This method might not be used in practice, but needs to exist for the interface
+    // If we reach here, it might be an error or a generic fallback
+    throw new Error(`Unexpected EXPRESSION node type. Node: ${JSON.stringify(node)}`);
+  }
+
+  visitFragment(node: FragmentNode): string {
+    const { spans } = node;
+    return spans.map((span: string | SQLNode) => {
+      return typeof span === 'string' ? span : this.toString(span);
+    }).join('');
+  }
+
+  visitFromClause(node: FromClauseNode): string {
+    const { expr, alias, sample } = node;
+    const isQuery = expr.type === 'SELECT_QUERY' || expr.type === 'SET_OPERATION';
+    const isTableRef = expr.type === 'TABLE_REF';
+    const ref = isQuery ? `(${this.toString(expr)})` : `${this.toString(expr)}`;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const from = alias && !(isTableRef && (expr as any).table?.join('.') === alias)
+      ? `${ref} AS ${quoteIdentifier(alias)}`
+      : `${ref}`;
+    return `${from}${sample ? ` TABLESAMPLE ${this.toString(sample)}` : ''}`;
+  }
+
+  visitFunction(node: FunctionNode): string {
+    const { name, args } = node;
+    return `${name}(${this.mapToString(args).join(', ')})`;
+  }
+
+  visitIn(node: InOpNode): string {
+    const { expr, values } = node;
+    return `(${this.toString(expr)} IN (${this.mapToString(values).join(', ')}))`;
+  }
+
+  visitInterval(node: IntervalNode): string {
+    const { steps, name } = node;
+    return `INTERVAL ${steps} ${name}`;
+  }
+
+  visitJoinClause(node: JoinNode): string {
+    const { left, right, joinVariant, joinType, condition, using, sample } = node;
+    const variant = joinVariant === 'REGULAR' ? '' : `${joinVariant} `;
+    let type = '';
+    let cond = '';
+
+    if (joinVariant !== 'CROSS') {
+      type = joinType !== 'INNER' ? `${joinType} ` : '';
+      cond = condition ? ` ON ${condition}`
+        : using ? ` USING (${using?.join(', ')})`
+        : '';
+    }
+    const samp = sample ? ` USING SAMPLE ${sample}` : '';
+    return `${left} ${variant}${type}JOIN ${right}${cond}${samp}`;
+  }
```

Comment:
I don't know if this is a real issue or just a bug in GitHub's code highlighting, but is there a string escape mismatch here? It seems like that would've broken the build, which is why I'm skeptical that it's a real bug

<img width="499" height="264" alt="Image" src="https://github.com/user-attachments/assets/c4c6b53a-b637-4972-a297-308edccb725d" />